### ISSUE TITLE
JIT alloc assert "!s_pagePool.isPoolPage(page)" running corefx tests in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/coreclr#18777]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_corefx_innerloop_prtest/388/consoleText
```
Assert failure(PID 7036 [0x00001b7c], Thread: 6744 [0x1a58]): Assertion failed '!s_pagePool.isPoolPage(page)' in 'Xunit.Sdk.ExtensibilityPointFactory:Dispose()' (IL size 21)

    File: d:\j\workspace\x64_checked_w---43686a05\src\jit\alloc.cpp Line: 301
    Image: D:\j\workspace\x64_checked_w---43686a05\bin\tests\Windows_NT.x64.Checked\testhost\dotnet.exe
```

**Comment by jkotas** [comment]
Recently changed in #15025

cc @mikedn

**Comment by mikedn** [comment]
I'll take a look. Looks like another shutdown race, hrm.

**Comment by mikedn** [comment]
@jkotas @AndyAyersMS I have a fix in #18782 but if changes discussed in #3408 go ahead then the fix won't be needed anymore. Good thing actually, attempting to clean-up during shutdown while work is still going on is kind of odd. But the old implementation did this, or at least attempted.

**Comment by AndyAyersMS** [comment]
Might as well take this fix. I don't yet know how close we are to resolving #3408.

**Comment by jkotas** [comment]
Fixed by #3408.
